EIPS/eip-5793.md

```diff
@@ -0,0 +1,49 @@
+---
+eip: 5793
+title: eth/68 - Add tx type to tx announcement
+description: Adds the transaction type and transaction size to tx announcement messages in the wire protocol
+author: Marius van der Wijden (@MariusVanDerWijden)
+discussions-to: https://ethereum-magicians.org/t/eip-5793-eth-68-add-transaction-type-to-tx-announcement/11364
+status: Draft
+type: Standards Track
+category: Networking
+created: 2022-10-18
+requires: 2464, 2481, 4938
+---
+
+## Abstract
+
+The [Ethereum Wire Protocol](https://github.com/ethereum/devp2p/tree/master/caps/eth.md) defines request and response messages for exchanging data between clients. The `NewPooledTransactionHashes` message announces transactions available in the node. This EIP extends this announcement message such that beside the transaction hashes, the node sends the transaction types and their sizes (as defined in EIP-2718) as well.
+
+## Motivation
+
+The `NewPooledTransactionHashes` message announces transaction hashes, allowing the peer to selectively fetch transactions it does not yet have.
+
+With the upcoming EIP-4844, a new transaction type is introduced for blob transactions. Since these blob transactions are large, naively broadcasting them to `sqrt(peers)` could significantly increase bandwidth requirements. Adding the transaction type and the size to the announcement message will allow nodes to select which transactions they want to fetch and also allow them to load balance or throttle peers based on past behavior.
```

Comment:
```suggestion
[EIP-4844](./eip-4844.md) introduces a new transaction type for blob transactions. Since these blob transactions are large, naively broadcasting them to `sqrt(peers)` could significantly increase bandwidth requirements. Adding the transaction type and the size to the announcement message will allow nodes to select which transactions they want to fetch and also allow them to load balance or throttle peers based on past behavior.
```
- **[Blocker]** Link to EIP-4844
- **[Suggestion]** Rephrase